untrusted/urts/src/lib.rs

```diff
@@ -193,4 +276,38 @@ mod tests {
         let builder = EnclaveBuilder::new(b"").debug(true);
         assert_eq!(builder.debug as c_int, 1);
     }
+
+    #[test]
+    fn no_report_function_results_in_error() {
+        let enclave = EnclaveBuilder::new(ENCLAVE).create().unwrap();
+        let target_info = MaybeUninit::zeroed();
+        let target_info = unsafe { target_info.assume_init() };
+        assert_eq!(
+            enclave.create_report(Some(target_info)).unwrap_err(),
+            Error::NoReportFunction
+        );
+    }
+
+    #[test]
+    fn report_function_provides_report() {
+        let enclave = EnclaveBuilder::new(ENCLAVE)
+            .report_fn(Some(|enclave, target_info| {
+                Ok(test_enclave::enclave_report(**enclave, target_info)?)
+            }))
+            .create()
+            .unwrap();
+
+        let target_info = MaybeUninit::zeroed();
+        let target_info = unsafe { target_info.assume_init() };
+        let report = enclave.create_report(Some(&target_info)).unwrap();
```

Comment:
Why are we passing a zero-value `target_info` instead of `None`?
```suggestion
        let report = enclave.create_report(None).unwrap();
```
... or perhaps another test that passes None

Reply:
Added a test case for `None`.
I'll try to rethink the code there, it feels like I'm just exercising the code and not testing it per se. 
Perhaps longer term the test `report_fn` should just provide a means to say what `enclave` and `target_info` were passed to it.